rtCommon/bidsInterface.py

```diff
@@ -1,43 +1,292 @@
+at the cloud projectServer which is a stub to forward requests (started with dataRemote=True),
+and another at the control room computer, run as a service and with dataRemote=False.
 
-In addition to creating a subclassed RemoteExtensible, an instance of this object must also
-be added to the projectServerRPC.py file, such as exposed_bidsInterface. And a data service
-must run at the control room computer that has an instance of this class with dataRemote=False.
-
-The control room instance will be the actual instance and the projectServerRPC instance is
-a shell instance (dataRemote=True) that forwards request to the control room instance.
+When not using RPC, i.e. when the projectServer is run without --dataRemote, there will be only
+one instance of dataInterface, as part of the projectServer with dataRemote=False.
 """
-import time
+import os
+import glob
+import nibabel as nib
 from rtCommon.remoteable import RemoteableExtensible
+from rtCommon.bidsArchive import BidsArchive
+from rtCommon.bidsIncremental import BidsIncremental
+from rtCommon.bidsCommon import getDicomMetadata
+from rtCommon.imageHandling import convertDicomImgToNifti
+from rtCommon.dataInterface import DataInterface
+from rtCommon.errors import ValidationError, RequestError
 
 
 class BidsInterface(RemoteableExtensible):
     """
-    Provides functions for experimenter scripts
+    Provides functions for accessing remote or local BIDs data depending on if dateRemote flag
+    is set true or false.
+
+    If dataRemote=True, then the RemoteExtensible parent class takes over and forwards all
+    requests to a remote server via a callback function registered with the RemoteExtensible
+    object. In that case *none* of the methods below will be locally invoked.
+
+    If dataRemote=False, then the methods below will be invoked locally and the RemoteExtensible
+    parent class is inoperable (i.e. does nothing).
     """
     def __init__(self, dataRemote=False):
         """
         Args:
-            dataRemote (bool): Set to true for the shell instance that will forward requests.
+            dataRemote (bool): Set to true for a passthrough instance that will forward requests.
                                Set to false for the actual instance running remotely
         """
         super().__init__(isRemote=dataRemote)
         if dataRemote is True:
             return
-        # Other initialization here
-
-    # Example function
-    def echo(self, val):
-        msg = f"Echo: {val}"
-        print(msg)
-        return msg
-
-    def testMethod(self, *args, **kwargs):
-        print(f'received args: {args}, kwargs: {kwargs}')
-        res = [args, kwargs]
-        return res
+        # local version initialization here
+        # TODO - make multithread streams possible
+        #      - get next streamId in thread safe way
+        #      - cleanup stale streams
+        nextStreamId = 1
```

Comment:
Yes, with the current implementation it only supports one stream at a time. If a new stream is initialized it will overwrite the current stream, i.e. replace it in the streamMap that maps streamId to stream instance. This is a TODO to support multiple streams.